packages/dataviews/src/field-types/email.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * WordPress dependencies
+ */
+import { isEmail } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import type {
+	DataViewRenderFieldProps,
+	SortDirection,
+	ValidationContext,
+	Operator,
+} from '../types';
+import { renderFromElements } from '../utils';
+import { OPERATOR_IS_ANY, OPERATOR_IS_NONE } from '../constants';
+
+function sort( valueA: any, valueB: any, direction: SortDirection ) {
+	return direction === 'asc'
+		? valueA.localeCompare( valueB )
+		: valueB.localeCompare( valueA );
+}
+
+function isValid( value: any, context?: ValidationContext ) {
+	// TODO: this implicitly means the value is required.
```

Comment:
🤔

Reply:
It's from https://github.com/Automattic/wp-calypso/blob/df80a74865c07b87e77796cc61cfae77ab2bf78c/packages/dataviews/src/field-types/integer.tsx#L25 🙈

Reply:
I know I know, I guess I would love to understand what are the issues with this and how we plan to solve them. I don't like keeping TODO comments in code for this reason, we forget why we added it in the first place. Please feel free to ignore this comment :)

Reply:
Does it make sense to add the `require` property to the `ValidationContext` and the we can pass the `require: true` from the definition of the field? 🤔

Reply:
I didn't know much thinking about validation yet to be honest. I know @louwie17 and @gigitux might have better clarity in this area.

Reply:
Can we keep that for its dedicated PR too and continue doing what we do in other field types. I want to keep this PR just about the new email type so we can ship it.